be/src/storage/vectorized/aggregate_iterator.h

```diff
@@ -3,7 +3,9 @@
 #pragma once
 
 #include "storage/vectorized/chunk_iterator.h"
+#include "storage/vectorized/row_source_mask.h"
 
 namespace starrocks::vectorized {
-ChunkIteratorPtr new_aggregate_iterator(ChunkIteratorPtr child, int factor = 0);
+ChunkIteratorPtr new_aggregate_iterator(ChunkIteratorPtr child, int factor = 0, bool is_vertical_merge = false,
+                                        bool is_key = false, std::vector<RowSourceMask>* source_masks = nullptr);
```

Comment:
Instead of passing `source_masks` here as an initialization parameter, an alternative way is to add a new interface in `ChunkIterator`, with a `std::vector<RowSourceMask>*` parameter along with `Chunk*`, like this:
```c++
class ChunkIterator {
public:
    virtual get_next(Chunk* chunk, std::vector<RowSourceMask>* chunk_sources) {
        return Status::NotSupported("get chunk with sources not supported by default");
    }
};
```
and implement this method in `HeapMeargeIterator`.

I know the latter one requires more work than the current one,  but do you think the latter one is easier to understand?

Reply:
done